templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,313 @@
+      nmcli c add type ovs-bridge conn.interface br-ex con-name br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac}
+    fi
+
+    old_conns=()
+    if "$BONDING"; then
+      iface=bond0
+      # store old connections for bringing down later
+      for bond_iface in ${bonded_ifaces[@]}; do
+        old_conns+=($(nmcli --fields UUID,DEVICE conn show --active | grep ${bond_iface} | awk '{print $1}'))
+      done
+    else
+      old_conns+=($(nmcli --fields UUID,DEVICE conn show --active | grep ${iface} | awk '{print $1}'))
+    fi
+
+    # find default port to add to bridge
+    if ! nmcli connection show ovs-port-phys0 &> /dev/null; then
+      nmcli c add type ovs-port conn.interface ${iface} master br-ex con-name ovs-port-phys0 ${phys_port_opts}
+    fi
+
+    if ! nmcli connection show ovs-port-br-ex &> /dev/null; then
+      nmcli c add type ovs-port conn.interface br-ex master br-ex con-name ovs-port-br-ex
+    fi
+
+    # bring down any old iface or bond connection (used for upgrading from local gw to shared gw mode)
+    for old_conn in ${old_conns[@]}; do
+      nmcli conn down $old_conn
+    done
+
+    if "$BONDING"; then
+      # Attach each interface as a slave to the bond ovs port
+      for bond_iface in ${bonded_ifaces[@]}; do
+        if ! nmcli connection show ovs-if-${bond_iface} &> /dev/null; then
+          nmcli c add type 802-3-ethernet conn.interface ${bond_iface} master ovs-port-phys0 con-name ovs-if-${bond_iface} \
+            connection.autoconnect-priority 100 802-3-ethernet.mtu ${iface_mtu}
+        fi
+      done
+    else
+      if ! nmcli connection show ovs-if-phys0 &> /dev/null; then
+        nmcli c add type 802-3-ethernet conn.interface ${iface} master ovs-port-phys0 con-name ovs-if-phys0 \
+          connection.autoconnect-priority 100 802-3-ethernet.mtu ${iface_mtu}
+      fi
+    fi
+
+    if ! nmcli connection show ovs-if-br-ex &> /dev/null; then
+      nmcli c add type ovs-interface slave-type ovs-port conn.interface br-ex master ovs-port-br-ex con-name \
+        ovs-if-br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac}
+    fi
+
+    # wait for DHCP to finish, verify connection is up
+    counter=0
+    while [ $counter -lt 5 ]; do
+      sleep 5
+      # check if connection is active
+      if nmcli --fields GENERAL.STATE conn show ovs-if-br-ex | grep -i "activated"; then
+        echo "OVS successfully configured"
+        ip a show br-ex
+        exit 0
+      fi
+      counter=$((counter+1))
+    done
+
+    echo "WARN: OVS did not succesfully activate NM connection. Attempting to bring up connections"
+    counter=0
+    while [ $counter -lt 5 ]; do
+      if nmcli conn up ovs-if-br-ex; then
+        echo "OVS successfully configured"
+        ip a show br-ex
+        exit 0
+      fi
+      sleep 5
+      counter=$((counter+1))
+    done
+
+    echo "ERROR: Failed to activate ovs-if-br-ex NM connection"
```

Comment:
Isn't this roughly what you would partially do for `ExecStop`?

Reply:
If so, consider refactoring this as a separate bash function which can be reused when the `ExecStop` is implemented.

Reply:
Yeah will do in a follow up